index.bs

```diff
@@ -13629,6 +13800,250 @@ internal method as follows.
+        1.  Return |esValue|.
+    1.  Return [=?=] |O|.\[[Get]](|P|, |Receiver|).
+</div>
+
+<h4 id="es-observable-array-getOwnPropertyDescriptor"><code>getOwnPropertyDescriptor</code></h4>
+
+<div algorithm="observable array exotic object getOwnPropertyDescriptor trap">
+    The steps for the <code>getOwnPropertyDescriptor</code> proxy trap for
+    [=observable array exotic objects=], given |O| and |P|, are as follows:
+
+    1.  Let |handler| be the <emu-val>this</emu-val> value.
+    1.  Let |length| be |handler|.\[[BackingList]]'s [=list/size=].
+    1.  If |P| is "length", then return [=!=]
+        [$FromPropertyDescriptor$](PropertyDescriptor{\[[Configurable]]: <emu-val>false</emu-val>,
+        \[[Enumerable]]: <emu-val>false</emu-val>, \[[Writable]]: <emu-val>true</emu-val>,
+        \[[Value]]: |length| }).
+    1.  If |P| [=is an array index=], then
+        1.  Let |index| be [=!=] [$ToUint32$](|P|).
+        1.  If |index| ≥ |length|, then return <emu-val>undefined</emu-val>.
+        1.  Let |esValue| be the result of [=converted to an ECMAScript value|converting=]
+            |handler|.\[[BackingList]][|index|] to an ECMAScript value.
+        1.  Assert: the above step never throws an exception.
+        1.  Return [=!=] [$FromPropertyDescriptor$](PropertyDescriptor{\[[Configurable]]:
+            <emu-val>true</emu-val>, \[[Enumerable]]: <emu-val>true</emu-val>, \[[Writable]]:
+            <emu-val>true</emu-val>, \[[Value]]: |esValue| }).
+    1.  Return [=!=] [$FromPropertyDescriptor$]([=?=] |O|.\[[GetOwnProperty]](|P|)).
+</div>
+
+<h4 id="es-observable-array-has"><code>has</code></h4>
+
+<div algorithm="observable array exotic object has trap">
+    The steps for the <code>has</code> proxy trap for
+    [=observable array exotic objects=], given |O| and |P|, are as follows:
+
+    1.  Let |handler| be the <emu-val>this</emu-val> value.
+    1.  If |P| is "length", then return <emu-val>true</emu-val>.
+    1.  If |P| [=is an array index=], then:
+        1.  Let |index| be [=!=] [$ToUint32$](|P|).
+        1.  If |index| &lt; |handler|.\[[BackingList]]'s [=list/size=], then return
+            <emu-val>true</emu-val>.
+        1.  Return <emu-val>false</emu-val>.
+    1.  Return [=?=] |O|.\[[HasProperty]](|P|).
+</div>
+
+<h4 id="es-observable-array-ownKeys"><code>ownKeys</code></h4>
+
+<div algorithm="observable array exotic object ownKeys trap">
+    The steps for the <code>ownKeys</code> proxy trap for
+    [=observable array exotic objects=], given |O|, are as follows:
+
+    1.  Let |handler| be the <emu-val>this</emu-val> value.
+    1.  Let |length| be |handler|.\[[BackingList]]'s [=list/size=].
+    1.  Let |keys| be an empty [=list=].
+    1.  Let |i| be 0.
+    1.  [=While=] |i| &lt; |length|:
+        1.  [=list/Append=] [=!=] [$ToString$](|i|) to |keys|.
+        1.  Set |i| to |i| + 1.
+    1.  [=list/Extend=] |keys| with [=!=] |O|.\[[OwnPropertyKeys]]().
```

Comment:
Aha, here the "define a random index with undefined as value on the target array" bit would be observable.  So yes, I think we should prevent defining it in `defineProperty` by either no-opping or throwing or treating no `[[Value]]` as undefined and doing the usual IDL type coercion (e.g. converting to 0 for numeric types).